### ISSUE TITLE
Open/Hide arbitrary files

### ISSUE THREAD

**Issue by badosu** [LMMS/lmms#1504]
So, occasionally I am browsing through my samples and I find some `txt` of `pdf` files related to them and I want to check about it.

At least for linux and OSX it would be very straightforward to launch an application to open files that are not opened natively on lmms (as opposed to .wav, .ogg, .xpf). For example, it's very common to have `xdg-open` available on linux distros. On OSX you have `open`.

Anyway, if this is not desirable as a lmms feature we could at least hide all files that can't be opened by it since they pollute the file tree.

Use case: I would like to check a sample license file if it's possible, if not it's just an useless icon.

**Comment by Sti2nd** [comment]
> Agreed, although a readme.txt or license.txt enhancement wouldn't be an
unreasonable request IMO

Were you thinking about opening them with the notepad? Maybe that would work actually :high_brightness:

**Comment by tresf** [comment]
> Were you thinking about opening them with the notepad? Maybe that would work actually :high_brightness:

I would never mention that application by name in a feature request, but yes, the `default text editor` would be invoked via the system's file association configuration.  On Ubuntu, this tends to be `gedit`, Mac is `TextEdit`, KDE is `Kate` (iirc), Windows is `notepad` :+1: .  But we wouldn't really need to know or care about that, we'd just let C++ open the default text editor. :smile:

**Comment by Sti2nd** [comment]
> I would never mention that application by name in a feature request

Darnit, totally wrong written. I meant Project Notes.

**Comment by tresf** [comment]
> Darnit, totally wrong written. I meant Project Notes.

Oh, you'd like to preview it internally?  I guess that is possible, but we would want to consider implementing a dedicated widget for it.

**Comment by Sti2nd** [comment]
> Oh, you'd like to preview it internally?

Yes, that would be cool, when you want to make a commercial track for example, you can then quicky open the licence.txt file to find out if that sample, vst, ladspa can be used.
